### PR TITLE
refactor: move express server out of docker

### PR DESCRIPTION
**Pull request by elbotho**
make sure to update your `.env` according to the template

```sh
yarn
#yarn dev-prep
yarn dev
```

Notes:
- I'm not sure it that is a widespread rule but many repos use the `name:action` schema inside package.json only if running `name` actually runs all `name:*` actions.
- We could combine `dev-prep` `dev` into one command but then the logs get pretty messy. That's why I chose to split it up.
- Not a super strong opinion on both points, open to changing it :) 
- `dotenvx run --env-file=.env -- ` is the same as  `dotenvx run -- `

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,11 @@
     "build": "npm-run-all --parallel \"build:*\"",
     "build:backend": "esbuild ./src/backend/index.ts --bundle --platform=node --outfile=dist/backend/index.cjs",
     "build:frontend": "vite build",
-    "dev": "docker compose up --watch --build",
-    "dev:edusharing": "dotenvx run --env-file=.env -- tsx --watch src/edusharing-mock/start-server.ts",
+    "dev-prep": "npm-run-all --parallel --parallel \"dev-prep:*\"",
+    "dev-prep:frontend": "vite build --watch",
+    "dev-prep:databases": "docker compose up --build",
```
Comment (LarsTheGlidingSquirrel) [inline package.json]:
```suggestion
    "dev-prep:databases": "docker compose up --detach --build",
```
I never looked at the logs from mongodb or mysql in all this time. I think we can suppress them to remove some noise from the console.

**Comment by LarsTheGlidingSquirrel** [comment]
Some files could be removed: 
```
.dockerignore
Dockerfile
Dockerfile.dev
```

**Comment by LarsTheGlidingSquirrel** [comment]
Health check in `docker-compose.yml` is no longer needed. Instead we might need to wait for the databases to be ready at some other place. Maybe another yarn command like `wait-for-server`?
https://github.com/serlo/serlo-editor-as-lti-tool/blob/72a9d67efd366f01152acb7db117fbfb1dfd5460/docker-compose.yml#L15

File: package.json [hunk 2]
```diff
@@ -18,9 +21,9 @@
     "lint:eslint": "eslint",
     "lint:prettier": "prettier --check .",
     "lint:tsc": "tsc --noEmit",
-    "start:dev": "tsx --watch src/backend/index.ts",
-    "start": "dotenvx run --env-file=.env -- node dist/backend/index.cjs",
-    "test": "dotenvx run --env-file=.env -- codeceptjs run --config e2e/codecept.conf.ts",
+    "start": "dotenvx run -- node dist/backend/index.cjs",
+    "start-ci": "dotenvx run -- nohup node dist/backend/index.cjs &",
```
Comment (hejtful) [inline package.json]:
```suggestion
```

**Review by hejtful — APPROVED** [review]
Very nice, thank you!

**Comment by elbotho** [comment]
Thank you for your feedback. Following the suggestion by @LarsTheGlidingSquirrel I tried detaching the db and run everything on `yarn dev` (no more `yarn dev-prep`). Maybe you can check if that also works for you and then we can merge 👍

File: package.json (hunk 1 above)
Comment (elbotho) [reply package.json]: NIce, thank you. See my comment below.

**Comment by hejtful** [comment]
Works fine for me! I'm honestly surprised that it works with `--parallel`, I would have expected the `dev:backend` to be faster than `dev:databases` and fail because Mongo isn't ready. But maybe I just don't understand the code that well 😁

**Comment by elbotho** [comment]
> Works fine for me! I'm honestly surprised that it works with --parallel, I would have expected the dev:backend to be faster than dev:databases and fail because Mongo isn't ready. But maybe I just don't understand the code that well 😁

Honestly: Same. I was working on some complex waiting logic and then just tried this and it worked fine. Let's see if it's reliable 🌳

**Review by LarsTheGlidingSquirrel — APPROVED** [review]
Thank you! :green_heart: 
When I run `yarn dev` the database is ready quickly enough. Let's see, we can add a wait later.

**Pull request merged**